packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3377,17 +3381,13 @@ export class ContainerRuntime
 					this.updateDocumentDirtyState(this.pendingMessagesCount !== 0);
 				}
 			}),
-			commitChanges: exitStagingMode(() => {
-				// All staged changes are in the PSM, so just replay them (ignore pre-staging batches)
-				if (this.connected) {
-					this.pendingStateManager.replayPendingStates(true /* onlyStagedBatched */);
-				} else {
-					// TODO:AB#37788: Refactor interplay with pending state manager and staging mode so that even in this case we end up
-					// squashing the ops submitted while in staging mode. This does not happen now due to the following clear without asking
-					// data stores to resubmit.
-					this.pendingStateManager.clearStagingFlags();
-				}
-			}),
+			commitChanges: (options = defaultStagingCommitOptions) =>
```

Comment:
Right now there is only a single option, so this works, but if we even had more options the behavior could change. i think it would be better to merge these options objects `{...defaultStagingCommitOptions, options}`, so default for unset values don't change if any value is set

Reply:
thanks, good catch.